File: BarberMe/BarberMe.Database/Models/RecommendationFeedback.cs
namespace BarberMe.Database.Models
{
    public class RecommendationFeedback
    {
        public int RecommendationFeedbackId { get; set; }

        public int RecommendationId { get; set; }
        public Recommendation Recommendation { get; set; } = null!;

        public int Rating { get; set; }
        public string? Comment { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: barberme_mobile/lib/models/recommendation_feedback.dart
class RecommendationFeedback {
  final int recommendationId;
  final int rating;
  final String? comment;
  final DateTime createdAt;

  RecommendationFeedback({
    required this.recommendationId,
    required this.rating,
    required this.comment,
    required this.createdAt,
  });

  factory RecommendationFeedback.fromJson(
    Map<String, dynamic> json,
  ) {
    return RecommendationFeedback(
      recommendationId:
          json['recommendationId'] as int,
      rating: json['rating'] as int,
      comment: json['comment']?.toString(),
      createdAt: DateTime.parse(
        json['createdAt'].toString(),
      ),
    );
  }
}